recipes/dpp/all/conanfile.py

```diff
@@ -0,0 +1,80 @@
+import os
+from conan import ConanFile
+from conan.tools.build import check_min_cppstd
+from conan.tools.cmake import CMakeToolchain, CMake, cmake_layout, CMakeDeps
+from conan.tools.scm import Git
+from conan.tools.files import download, unzip
+
+required_conan_version = ">=1.65"
+
+class DPPConan(ConanFile):
+    name = "dpp"
+    license = "Apache-2.0"
+    package_type = "shared-library"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/brainboxdotcc/DPP"
+    description = "D++ is a lightweight and efficient library for Discord"
+    topics = ("discord")
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"shared": [True, False]}
+    default_options = {"shared": True}
+
+    @property
+    def _min_cppstd(self):
+        return 17
+
+    def validate(self):
+        if self.settings.compiler.get_safe("cppstd"):
+            check_min_cppstd(self, self._min_cppstd)
+
+    def requirements(self):
+        self.requires("nlohmann_json/3.11.2")
+        self.requires("openssl/3.1.2")
+        self.requires("zlib/1.3")
```

Comment:
```suggestion
        self.requires("openssl/[>=1.1 <4]")
        self.requires("zlib/[>=1.2.11 <2]")
```

Some libraries can be made version ranges as per https://github.com/conan-io/conan-center-index/blob/master/docs/adding_packages/dependencies.md#version-ranges :)